assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * DashboardIdeasWidget component
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Tab from '@material/react-tab';
+import TabBar from '@material/react-tab-bar';
+import { useHash, useMount } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useRef, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import whenActive from '../../../../../util/when-active';
+import NewIdeas from './NewIdeas';
+import SavedIdeas from './SavedIdeas';
+import DraftIdeas from './DraftIdeas';
+
+const getHash = ( hash ) => hash ? hash.replace( '#', '' ) : false;
+const isValidHash = ( hash ) => getHash( hash ) in DashboardIdeasWidget.tabToIndex;
+
+const DashboardIdeasWidget = () => {
+	const ideaHubContainer = useRef();
+	const [ hash, setHash ] = useHash();
+	const [ activeTabIndex, setActiveTabIndex ] = useState( DashboardIdeasWidget.tabToIndex[ getHash( hash ) ] );
+	const activeTab = DashboardIdeasWidget.tabIDsByIndex[ activeTabIndex ];
+
+	const getIdeaHubContainerOffset = () => {
+		const tabletBreakpoint = global.window.matchMedia( '(min-width: 600px)' );
+		const desktopBreakpoint = global.window.matchMedia( '(min-width: 960px)' );
+		let adminBarHeight = 0;
+		let siteHeaderHeight = 68;
+		let marginBottom = 16;
+
+		if ( tabletBreakpoint.matches ) {
+			adminBarHeight = 46;
+		}
+
+		if ( desktopBreakpoint.matches ) {
+			siteHeaderHeight = 84;
+			marginBottom = 24;
+			adminBarHeight = 32;
+		}
```

Comment:
Instead of hardcoding all these numbers, we should use actual heights of appropriate elements:

```
siteHeaderHeight = document.querySelector('.googlesitekit-header')?.offsetHeight || 0;
adminBarHeight = document.getElementById( 'wpadminbar' )?.offsetHeight || 0;
```

Not sure about marginBottom, do we really need it? :thinking:

Reply:
I was on the fence about hardcoding those values since they will never change, but better be safe and make them dynamic. Updated. As for the margin, it's just to have some spacing between the header and the widget.